HAZIK/Fenyo/HF5/Feny_HF5.py

```diff
@@ -0,0 +1,256 @@
+    for i in range(15,41):
+        # minden életkorra kilistázzuk az összes olyan életkorú embert, és nekik kilistázzuk az összes ismerősük életkorait
+        ismerosok_korai = sum([[subG.nodes[node]['age'] for node in list(subG.neighbors(x))] for x in [node for node in subG.nodes if subG.nodes[node]['age'] == i]], [])
+        # Ezekből az életkorokból kiszűrjük a nullákat, majd a 15-40 év közötti tartományt eltároljuk a kormátrixban
+        kormatrix[i] = pd.Series([kor for kor in ismerosok_korai if kor != 0]).value_counts(bins = np.linspace(15,41,27), dropna = False).sort_index().values
+        
+    sns.heatmap([kormatrix[elem] for elem in kormatrix], square = True, ax = axes[0,0])
+    axes[0,0].set(title = 'Korcsoportok közötti kapcsolatok száma')
+    axes[0,0].set_xticklabels(np.linspace(15,41,13, dtype = int))
+    axes[0,0].set_yticklabels(np.linspace(15,41,13, dtype = int), rotation = 0)
+    
+    
+    # Férfiak közötti kapcsolatok
+    kormatrix = {}
+    # végigmegyünk a vizsgált életkorokon
+    for i in range(15,41):
+        # minden életkorra kilistázzuk az összes olyan életkorú FÉRFI nemű embert, és nekik kilistázzuk az összes FÉRFI ismerősük életkorait
+        ismerosok_korai = sum([[subG.nodes[node]['age'] for node in list(subG.neighbors(x)) if subG.nodes[node]['gender'] == 'male'] for x in [nood for nood in [node for node in subG.nodes if subG.nodes[node]['age'] == i] if subG.nodes[nood]['gender'] == 'male']], [])
+        # Ezekből az életkorokból kiszűrjük a nullákat, majd a 15-40 év közötti tartományt eltároljuk a kormátrixban
+        kormatrix[i] = pd.Series([kor for kor in ismerosok_korai if kor != 0]).value_counts(bins = np.linspace(15,41,27), dropna = False).sort_index().values
+        
+    sns.heatmap([kormatrix[elem] for elem in kormatrix], square = True, ax = axes[0,1])
+    axes[0,1].set(title = 'Férfiak közötti kapcsolatok száma')
+    axes[0,1].set_xticklabels(np.linspace(15,41,13, dtype = int))
+    axes[0,1].set_yticklabels(np.linspace(15,41,13, dtype = int), rotation = 0)
+    
+    
+    # Nők közötti kapcsolatok
+    kormatrix = {}
+    # végigmegyünk a vizsgált életkorokon
+    for i in range(15,41):
+        # minden életkorra kilistázzuk az összes olyan életkorú NŐI nemű embert, és nekik kilistázzuk az összes NŐI ismerősük életkorait
+        ismerosok_korai = sum([[subG.nodes[node]['age'] for node in list(subG.neighbors(x)) if subG.nodes[node]['gender'] == 'female'] for x in [nood for nood in [node for node in subG.nodes if subG.nodes[node]['age'] == i] if subG.nodes[nood]['gender'] == 'female']], [])
+        # Ezekből az életkorokból kiszűrjük a nullákat, majd a 15-40 év közötti tartományt eltároljuk a kormátrixban
+        kormatrix[i] = pd.Series([kor for kor in ismerosok_korai if kor != 0]).value_counts(bins = np.linspace(15,41,27), dropna = False).sort_index().values
+        
+    sns.heatmap([kormatrix[elem] for elem in kormatrix], square = True, ax = axes[1,0])
```

Comment:
a heatmap kreálás ugyanúgy ismétlődés, ezt meg lehetne oldani negyedennyi sorból is